src/libraries/Common/src/System/Collections/Generic/LargeArrayBuilder.SpeedOpt.cs

```diff
@@ -42,10 +38,10 @@ public LargeArrayBuilder(bool initialize)
         /// Do not add more than <paramref name="maxCapacity"/> items to this builder.
         /// </remarks>
         public LargeArrayBuilder(int maxCapacity)
-            : this()
         {
             Debug.Assert(maxCapacity >= 0);
 
+            this = default;
```

Comment:
is there any functional or perf difference between assignment like this vs `: this()`? If not, I personally find old syntax clearer, do we change entire code base with this?

Reply:
Oh,  I just realized you meant this = default va a this ctor delegation. Similar answer: it depends on whether a ctor is defined as to whether or zero initializes or invokes a ctor.